File: tutorial/codelab/merge_sorted_array/merge_sorted_array.dart
void main() {
  List<int> arr = [1, 3, 7, 9, 11, 13, 17, 23];
  List<int> arr2 = [2, 4, 6, 8, 10];

  solution1(arr, arr2);
}

void solution1(arr, arr2) {
  int n = arr.length;
  int m = arr2.length;
  int i = 0;
  int j = 0;
  int k = 0;
  List<int> data = [];

  while (i < n && j < m) {
    if (arr[i] < arr2[j]) {
      data.insert(k++, arr[i++]);
    } else {
      data.insert(k++, arr2[j++]);
      // data.add(arr2[j++]);
    }
  }

  while (i < n) {
    data.insert(k++, arr[i++]);
  }
  while (j < m) {
    data.insert(k++, arr2[j++]);
  }

  print(data);
}

File: tutorial/codelab/merge_sorted_array/merge_sorted_array.cpp
#include <iostream>
using namespace std;

void solution1(int arr[], int arr2[], int n, int m)
{
    int len = n + m;
    int mixArr[len] = {0};
    int a = 0;
    int i = 0;
    int j = 0;
    while (i < n && j < m)
    {
        if (arr[i] < arr2[j])
        {
            mixArr[a++] = arr[i++];
        }
        else
        {

            mixArr[a++] = arr2[j++];
        }
    }

    while (i < n)
    {
        mixArr[a++] = arr[i++];
    }
    while (j < m)
    {
        mixArr[a++] = arr2[j++];
    }

    for (int k = 0; k < len; k++)
    {
        cout << mixArr[k] << " ";
    }
}
int main()
{
    int arr[8] = {
        1,
        3,
        7,
        9,
        11,
        13,
        17,
        23};
    int arr2[7] = {
        2,
        4,
        6,
        8,
        10,
        20, 24};

    solution1(arr, arr2, 8, 7);
}
